pkg/kubelet/cm/container_manager.go

```diff
@@ -36,6 +36,16 @@ type ContainerManager interface {
 
 	// Returns internal Status.
 	Status() Status
+
+	// NewPodContainerManager is a factory method which returns a podContainerManager object
+	// Returns a noop implementation if qos cgroup hierarchy is not enabled
+	NewPodContainerManager() PodContainerManager
+
+	// GetMountedSubsystems returns the mounted cgroup subsytems on the node
+	GetMountedSubsystems() *CgroupSubsystems
```

Comment:
it feels like the factoring could be better here.  it feels like reducing memory limits should be handled in a module opaque to the kubelet.